js_modules/dagster-ui/packages/ui-core/src/assets/RecentUpdatesTimeline.tsx

```diff
@@ -0,0 +1,209 @@
+      start: number;
+      end: number;
+      materializations: typeof materializations;
+    }> = [];
+
+    firstPassBucketsArray.forEach((bucket) => {
+      const lastBucket = secondPassBucketsArray[secondPassBucketsArray.length - 1];
+      if (!lastBucket || lastBucket.end !== bucket.number - 1) {
+        secondPassBucketsArray.push({
+          start: bucket.number,
+          end: bucket.number,
+          materializations: bucket.materializations,
+        });
+      } else {
+        lastBucket.end = bucket.number;
+        lastBucket.materializations = [...lastBucket.materializations, ...bucket.materializations];
+      }
+    });
+    return secondPassBucketsArray;
+  }, [viewport.width, buckets, materializations, startTimestamp, range]);
+
+  const formatDateTime = useFormatDateTime();
+
+  return (
+    <Box flex={{direction: 'column', gap: 4}}>
+      <Box flex={{direction: 'row', justifyContent: 'space-between'}}>
+        <Subtitle2>Recent updates</Subtitle2>
+        <Caption color={Colors.textLighter()}>
+          {materializations.length === 100
+            ? 'Last 100 updates'
+            : `Showing all ${materializations.length} updates`}
+        </Caption>
+      </Box>
+      <Box border="all" padding={6 as any} style={{height: 32}}>
+        <div {...containerProps} style={{width: '100%', height: 20, position: 'relative'}}>
+          {bucketedMaterializations.map((bucket) => {
+            const width = tickWidth * (bucket.end - bucket.start + 1);
+            return (
+              <TickWrapper
+                key={bucket.start}
+                style={{
+                  left: tickWidth * bucket.start,
+                  width,
+                }}
+              >
+                <Popover
+                  content={
+                    <Box flex={{direction: 'column', gap: 8}}>
+                      <Box padding={8} border="bottom">
+                        <Subtitle2>Materializations</Subtitle2>
+                      </Box>
+                      <div style={{maxHeight: 'min(80vh, 300px)', overflow: 'scroll'}}>
+                        {bucket.materializations
+                          .sort((a, b) => parseInt(b.timestamp) - parseInt(a.timestamp))
+                          .map((materialization, index) => (
+                            <AssetUpdate assetKey={assetKey} event={materialization} key={index} />
+                          ))}
+                      </div>
+                    </Box>
+                  }
+                >
+                  <Tick style={{width}}>
+                    {bucket.materializations.length > 1 ? bucket.materializations.length : null}
+                  </Tick>
+                </Popover>
+              </TickWrapper>
+            );
+          })}
+        </div>
+      </Box>
+      <Box padding={{top: 4}} flex={{justifyContent: 'space-between'}}>
+        <Caption color={Colors.textLighter()}>
+          {formatDateTime(new Date(startTimestamp), {})}
```

Comment:
Would `<TimestampDisplay>` or `<Timestamp>` work? Usually `formatDateTime` would be for some unusual rendering where we need the options object.